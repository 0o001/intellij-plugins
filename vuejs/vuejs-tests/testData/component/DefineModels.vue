<script setup>
let modelValue = defineModel({required: true, type: String})
let count = defineModel('count', {type: Number})
let countLocal = defineModel('countLocal', {local: true, type: Number})
let countRequired = defineModel('countRequired', {required: true, type: [Number, Boolean]})
let countType = defineModel('countType', {required: false, type: String})
let countAny = defineModel('countAny')
</script>

<template></template>