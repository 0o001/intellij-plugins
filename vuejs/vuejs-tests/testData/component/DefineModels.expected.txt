{
 class: "VueSourceComponent",
 components: {
  DefineModels: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueScriptSetupModelInputProperty",
   name: "count",
   jsType: "number",
   required: false,
  },
  {
   class: "VueScriptSetupModelInputProperty",
   name: "countAny",
   jsType: "any",
   required: false,
  },
  {
   class: "VueScriptSetupModelInputProperty",
   name: "countLocal",
   jsType: "number",
   required: false,
  },
  {
   class: "VueScriptSetupModelInputProperty",
   name: "countRequired",
   jsType: "number | boolean",
   required: true,
  },
  {
   class: "VueScriptSetupModelInputProperty",
   name: "countType",
   jsType: "string",
   required: false,
  },
  {
   class: "VueScriptSetupModelInputProperty",
   name: "modelValue",
   jsType: "string",
   required: true,
  },
 ],
 emits: [
  {
   class: "VueScriptSetupModelEvent",
   name: "update:count",
   eventJSType: "(value: number) => void",
   params: [
    {
     name: "value",
     type: "number",
    },
   ],
   hasStrictSignature: true,
   callSignature: "(event: "update:count", value: number) => void",
  },
  {
   class: "VueScriptSetupModelEvent",
   name: "update:countAny",
   eventJSType: "(value: any) => void",
   params: [
    {
     name: "value",
     type: "any",
    },
   ],
   hasStrictSignature: true,
   callSignature: "(event: "update:countAny", value: any) => void",
  },
  {
   class: "VueScriptSetupModelEvent",
   name: "update:countLocal",
   eventJSType: "(value: number) => void",
   params: [
    {
     name: "value",
     type: "number",
    },
   ],
   hasStrictSignature: true,
   callSignature: "(event: "update:countLocal", value: number) => void",
  },
  {
   class: "VueScriptSetupModelEvent",
   name: "update:countRequired",
   eventJSType: "(value: number | boolean) => void",
   params: [
    {
     name: "value",
     type: "number | boolean",
    },
   ],
   hasStrictSignature: true,
   callSignature: "(event: "update:countRequired", value: number | boolean) => void",
  },
  {
   class: "VueScriptSetupModelEvent",
   name: "update:countType",
   eventJSType: "(value: string) => void",
   params: [
    {
     name: "value",
     type: "string",
    },
   ],
   hasStrictSignature: true,
   callSignature: "(event: "update:countType", value: string) => void",
  },
  {
   class: "VueScriptSetupModelEvent",
   name: "update:modelValue",
   eventJSType: "(value: string) => void",
   params: [
    {
     name: "value",
     type: "string",
    },
   ],
   hasStrictSignature: true,
   callSignature: "(event: "update:modelValue", value: string) => void",
  },
 ],
}
